docs/devguide/flashfirmware.rst

```diff
@@ -1,58 +1,83 @@
+build, and additional data is added to the hex file after.
+
+Use yotta to build::
 
-Start the build with either yotta::
+  yt target bbc-microbit-classic-gcc@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc
 
-  yt build
+Run yotta update to fetch remote assets::
+
+  yt up
 
-...or use the Makefile::
+The yotta tool is used to build MicroPython, but before that takes place
+additional files have to be generated by the Makefile in preparation for the 
+build, and additional data is added to the hex file after.
 
-  make all
+Running the ``make all`` command executes the following steps:
 
-The result is a microbit-micropython hex file (i.e. ``microbit-micropython.hex``)
-found in the build/bbc-microbit-classic-gcc-nosd/source from the root of the
-repository.
+- The ``tools/makeversionhdr.py`` script creates the ``microbitversion.h`` file 
+  with macros containing build information
+- Yotta builds the source and creates a bare hex file with just the firmware
+- The ``tools/adduicr.py`` script adds the UICR to the bare hex
+- The final hex file is placed in ``build/firmware.hex``
+- The user can optionally append a script using ``tools/makecombinedhex.py`` 
+  (or other tools)
 
-The Makefile does some extra preprocessing of the source, which is needed only
-if you add new interned strings to ``qstrdefsport.h``. The Makefile also puts
-the resulting firmware at build/firmware.hex, and includes some convenience
-targets.
+micro:bit V2
+============
 
-Preparing firmware and a Python program
----------------------------------------
+This applies to MicroPython for the micro:bit V2, the source of which can be 
+found here: `microbit-foundation/micropython-microbit-v2 <https://github.com/microbit-foundation/micropython-microbit-v2>`_.
 
-tools/makecombined
+The repository also contains a history of 
+`MicroPython firmware builds <https://github.com/microbit-foundation/micropython-microbit-v2/actions>`_.
 
-hexlify
+This is a port of MicroPython to the micro:bit which uses CODAL as the 
+underlying target platform.
 
+After cloning this repository update the submodules::
 
+    $ git submodule update --init
 
-Flashing to the micro:bit
--------------------------
+Then build the MicroPython cross-compiler::
 
+    $ make -C lib/micropython/mpy-cross
 
+After setting up, go to the src/ directory and build::
 
+    $ cd src
 
-**Installation Scenarios**
+    $ make
 
-* :ref:`Windows <microbit-windows>`
-* :ref:`OS X <microbit-osx>`
-* :ref:`Linux <microbit-linux>`
-* :ref:`Debian and Ubuntu <microbit-debian-ubuntu>`
-* :ref:`Red Hat Fedora/CentOS <microbit-redhat>`
-* :ref:`Raspberry Pi <microbit-rpi>`
+That will build both ``libmicropython.a`` (from source in ``src/codal_port/``) and the 
+CODAL app (from source in ``src/codal_app/``). The resulting firmware will be 
+``MICROBIT.hex`` in the ``src/`` directory which can be copied to the micro:bit.
```

Comment:
We can probably just mention the hex and lib out the .a files.

Reply:
@microbit-mark this is still outstanding, but if you think we should leave it in let me know.

Reply:
I've just left the hex info in https://github.com/bbcmicrobit/micropython/commit/939efde1f837cd9aea60e972e3bb135b215baff6